src/pages/api/index.js

```diff
@@ -14,5 +15,12 @@ export default function handler(req, res) {
     return hasRoomName && hasAvailableSlot;
   });
 
-  return res.status(200).json(fliteredRooms);
+  const fliteredRoomsByCluster = fliteredRooms.filter((room) => {
+    const targetCluster =
+      cluster == "ALL" ? true : cluster_data[room] == cluster;
+
+    return targetCluster;
+  });
```

Comment:
Have you tried combining the new filter logic with the previous 'filteredRooms' logic?